src/cow-react/modules/advancedOrders/containers/AdvancedOrdersWidget/index.tsx

```diff
@@ -34,6 +39,22 @@ export function AdvancedOrdersWidget() {
   const actions = useAdvancedOrdersActions()
   const isTradePriceUpdating = useIsQuoteLoading()
 
+  // Number of parts
+  const { numberOfPartsError, numberOfPartsValue } = useAtomValue(advancedOrdersSettingsAtom)
+  const parseNumberOfParts = useParseNumberOfParts()
+
+  // Slippage
+  const [slippageInput, setSlippageInput] = useState('')
```

Comment:
<img width="1318" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/afd7a1ec-6aaa-41e3-a0e3-fc07ecd78e30">

In general, the logic should fit this diagram.
Basic principles:
1. `TradeNumberInput` returns only `number` or `null`
2. `TradeNumberInput` has `precision` a number of digits after comma (like [here](https://taiga-ui.dev/components/input-number/API))
3. Validation of slippage and numOfParts should be presented as a pure function because it's deterministic and depends only on `number | null` value
4. Since a validation error is deterministic, we don't need to store it. It always can be calculated on flight

@nenadV91

Reply:
`Since a validation error is deterministic`
I'm not sure about this, because we will probably use those errors for Buttons state (disabled/enabled) and I think its easier to just access it from the state instead of passing it down

Reply:
Why it's not deterministic? It supposed to be a pure function

Reply:
I would store it because it will be used in different components